.github/workflows/backend.yml

```diff
@@ -461,8 +461,28 @@ jobs:
           cache-dependency-path: requirements-dev-frozen.txt
           install-cmd: pip install -r requirements-dev-frozen.txt
 
-      - name: Run backend typing (${{ steps.setup.outputs.matrix-instance-number }} of ${{ strategy.job-total }})
-        run: make backend-typing
+      - run: make backend-typing
+        id: run
+
+      - uses: getsentry/action-github-app-token@97c9e23528286821f97fba885c1b1123284b29cc  # v2.0.0
+        id: token
+        continue-on-error: true
+        with:
+          app_id: ${{ vars.SENTRY_INTERNAL_APP_ID }}
+          private_key: ${{ secrets.SENTRY_INTERNAL_APP_PRIVATE_KEY }}
+
+      # only if `backend-typing` succeeds should we try and trim the blocklist
+      - run: |
+          python3 -m tools.mypy_helpers.make_module_ignores
+          git diff --exit-code
+
+      - name: apply blocklist changes
+        if: steps.token.outcome == 'success' && steps.run.outcome == 'success' && github.ref != 'refs/heads/master' && always()
```

Comment:
is `always()` needed here? and was the intent that this runs on non-master branches i.e. the blocklist is reviewed in PRs?

Reply:
the ~hope was that the blocklist only goes down.  this is the thing that pushes to PRs so it'll only run there.  the `always()` is needed so that when the blocklist-change thing fails (if it makes a change) it'll still run